<?xml version="1.0"?>
<launch>

<arg name="use_sim_time" default="true"/>
<arg name="odom_gt" default="true"/>
<arg name="global_planner_use" default="dubins"/>
<param name ="/use_sim_time" value="$(arg use_sim_time)"/>

  <arg name="no_static_map" default="true"/>
  <node name="ackermann_converter" pkg="align_gazebo" type="cmd_vel_to_ackermann_drive.py"/>
  <node name="gt_broadacaster" pkg="align_gazebo" type="tf_broadcaster_autoware.py"/>
  <!-- <arg name="base_global_planner" default="navfn/NavfnROS"/> -->
  <!-- <arg name="base_global_planner" default="dubins_planner/DubinsGlobalPlanner"/>  -->
  <arg name="base_global_planner" default="reeds_shepp_planner/ReedsSheppGlobalPlanner"/> 

  <!-- <param name="base_global_planner" value="dubins_planner/DubinsGlobalPlanner" />  -->
  <!-- <arg name="base_local_planner" default="dwa_local_planner/DWAPlannerROS"/> -->
  <arg name="base_local_planner" default="teb_local_planner/TebLocalPlannerROS"/>

  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

    <param name="base_global_planner" value="$(arg base_global_planner)"/>
    <param name="base_local_planner" value="$(arg base_local_planner)"/>  
    <!-- <rosparam file="$(find align_navigation)/config/planner.yaml" command="load"/> -->
    <rosparam file="$(find align_navigation)/config/try/teb_local_planner_params.yaml" command="load"/>
    

    <!-- observation sources located in costmap_common.yaml -->
    <rosparam file="$(find align_navigation)/config/costmap_common.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find align_navigation)/config/costmap_common.yaml" command="load" ns="local_costmap" />

    <!-- local costmap, needs size -->
    <rosparam file="$(find align_navigation)/config/costmap_local.yaml" command="load" ns="local_costmap" />
    <param name="local_costmap/width" value="10.0"/>
    <param name="local_costmap/height" value="10.0"/>

    
    <!-- global costmap with laser, for odom_navigation_demo -->
    <rosparam file="$(find align_navigation)/config/costmap_global_laser.yaml" command="load" ns="global_costmap" if="$(arg no_static_map)"/>
    <param name="align_navigation/width" value="100.0" if="$(arg no_static_map)"/>
    <param name="align_navigation/height" value="100.0" if="$(arg no_static_map)"/>
  </node>

</launch>